godot-core/src/builtin/rect2.rs

```diff
@@ -44,6 +44,180 @@ impl Rect2 {
         }
     }
 
+    /// Returns a rectangle with equivalent position and area, modified so that the top-left corner is the origin and `width` and `height` are positive.
+    ///
+    /// _Godot equivalent: `Rect2.abs()`_
+    #[inline]
+    pub fn abs(&self) -> Self {
+        Self {
+            position: self.position + Vector2::new(self.size.x.min(0.0), self.size.y.min(0.0)),
+            size: self.size.abs(),
+        }
+    }
+
+    /// Returns true if this rectangle (inclusively) encloses `b`. This is true when `self` covers all the area of `b`, and possibly (but not necessarily) more.
+    ///
+    /// _Godot equivalent: `Rect2.encloses(Rect2 b)`_
+    #[inline]
+    pub fn encloses(&self, b: Rect2) -> bool {
+        b.position.x >= self.position.x
+            && b.position.y >= self.position.y
+            && b.position.x + b.size.x <= self.position.x + self.size.x
+            && b.position.y + b.size.y <= self.position.y + self.size.y
+    }
+
+    /// Returns a copy of this rectangle expanded to include a given point.
+    ///
+    /// Note: This method is not reliable for `Rect2` with a negative size. Use [`abs`][Self::abs]
+    /// to get a positive sized equivalent rectangle for expanding.
+    ///
+    /// _Godot equivalent: `Rect2.expand(Vector2 to)`_
+    #[inline]
+    pub fn expand(&self, to: &Vector2) -> Self {
+        self.merge(&Rect2::new(*to, Vector2::ZERO))
+    }
+
+    /// Returns a larger rectangle that contains this `Rect2` and `b`.
+    ///
+    /// Note: This method is not reliable for `Rect2` with a negative size. Use [`abs`][Self::abs]
+    /// to get a positive sized equivalent rectangle for merging.
+    ///
+    /// _Godot equivalent: `Rect2.merge(Rect2 b)`_
+    #[inline]
+    pub fn merge(&self, b: &Self) -> Self {
+        let position = Vector2::new(
+            self.position.x.min(b.position.x),
+            self.position.y.min(b.position.y),
+        );
+        let end = Vector2::new(
+            (self.position.x + self.size.x).max(b.position.x + b.size.x),
+            (self.position.y + self.size.y).max(b.position.y + b.size.y),
+        );
+
+        Self {
+            position,
+            size: end - position,
+        }
```

Comment:
It probably makes sense to add a `Rect2::from_corners(position, end)` here, like we did for other types (was it `Rect2i` or `Aabb`?).